summaries/analysis/Analyzer.py

```diff
@@ -207,3 +217,213 @@ def _get_tokens(self, text: str) -> List[str]:
+        if train_set:
+            seen_samples_ref_train = set([sample[reference_text_column_name] for sample in train_set])
+            seen_samples_summary_train = set([sample[summary_text_column_name] for sample in train_set])
+        if validation_set:
+            seen_samples_ref_val = set([sample[reference_text_column_name] for sample in validation_set])
+            seen_samples_summary_val = set([sample[summary_text_column_name] for sample in validation_set])
+        if test_set:
+            seen_samples_ref_test = set([sample[reference_text_column_name] for sample in test_set])
+            seen_samples_summary_test = set([sample[summary_text_column_name] for sample in test_set])
+
+        # Now simply cross-check which elements occur in both.
+        # TODO: The re-checking of set existence looks a bit cumbersome
+        reference_leakage = 0
+        summary_leakage = 0
+        if train_set:
+            for ref_text in seen_samples_ref_train:
+                if validation_set:
+                    if ref_text in seen_samples_ref_val:
+                        reference_leakage += 1
+                        print(f"Reference leakage between train and validation set spotted:")
+                        print(f"{ref_text[:self.print_cutoff_length]}")
+                if test_set:
+                    if ref_text in seen_samples_ref_test:
+                        reference_leakage += 1
+                        print(f"Reference leakage between train and test set spotted:")
+                        print(f"{ref_text[:self.print_cutoff_length]}")
+
+            for summ_text in seen_samples_summary_train:
+                if validation_set:
+                    if summ_text in seen_samples_summary_val:
+                        summary_leakage += 1
+                        print(f"Reference leakage between train and validation set spotted:")
+                        print(f"{summ_text[:self.print_cutoff_length]}")
+                if test_set:
+                    if summ_text in seen_samples_summary_test:
+                        summary_leakage += 1
+                        print(f"Reference leakage between train and test set spotted:")
+                        print(f"{summ_text[:self.print_cutoff_length]}")
+
+        if validation_set:
+            for ref_text in seen_samples_ref_val:
+                if test_set:
+                    if ref_text in seen_samples_ref_test:
+                        reference_leakage += 1
+                        print(f"Reference leakage between validation and test set spotted:")
+                        print(f"{ref_text[:self.print_cutoff_length]}")
+
+        for summ_text in seen_samples_summary_val:
+            if test_set:
+                if summ_text in seen_samples_summary_test:
+                    summary_leakage += 1
+                    print(f"Reference leakage between validation and test set spotted:")
+                    print(f"{summ_text[:self.print_cutoff_length]}")
```

Comment:
This can be simplified with a list containing all valid splits, and then does not require the individual `if ...` checks anymore, plus simplifies to a single double-nested iteration without explicit references.